File: Números combinatorios/Java/Numero_Combinatorio_v2.java
package numero_combinatorio_v2;

import java.util.Scanner;

public class Numero_Combinatorio_v2 {

    public static long combinatorio(int n, int k) {
        if (n >= 0 && k == 0) {
            return 1;
        }
        if (n == 0 && k > 0) {
            return 0;
        }
        return combinatorio(n - 1, k - 1) + combinatorio(n - 1, k);
    }

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese n: ");
        int n = reader.nextInt();
        System.out.printf("Ingrese k: ");
        int k = reader.nextInt();

        if (n <= 0 || k <= 0) {
            System.out.printf("n y k deben ser mayores que cero%n");
        } else if (k > n) {
            System.out.printf("n debe ser mayor o igual que k%n");
        } else {
            System.out.printf("El combinatorio de n en k es %d%n", combinatorio(n, k));
        }
    }
}

File: Números combinatorios/Python/numero_combinatorio_v2.py
#Creado por Andrés Melgar
#para iterando++ http://iterando.online/
#entrada: http://iterando.online/numero-combinatorio-en-python/
def combinatorio(n, k):
    if n >= 0 and k == 0:
        return 1
    if n == 0 and k > 0:
        return 0
    return combinatorio(n - 1, k - 1) + combinatorio(n - 1, k)

n=int(input("Ingrese n: "))
k=int(input("Ingrese k: "))

if n < 0 or k < 0:
    print("n y k deben ser mayores o iguales que cero")
elif k > n:
    print("n debe ser mayor o igual que k")
else:
    print("El combinatorio de n en k es {:d}".format(combinatorio(n, k)))